src/sst/core/model/python/pymodel_unitalgebra.cc

```diff
@@ -0,0 +1,338 @@
+// operators end up being the same function.  Thus, the add, sub, mult
+// and div functions are used in both places.
+PyNumberMethods PyModel_UnitAlgebraNumMeth = {
+    (binaryfunc)unitAlgebraAdd,    // binaryfunc nb_add
+    (binaryfunc)unitAlgebraSub,    // binaryfunc nb_subtract
+    (binaryfunc)unitAlgebraMul,    // binaryfunc nb_multiply
+    SST_NB_DIVIDE((binaryfunc)unitAlgebraDiv)    // binaryfunc nb_divide, py2 only
+    nullptr,                // binaryfunc nb_remainder
+    nullptr,                // binaryfunc nb_divmod
+    nullptr,                // ternaryfunc nb_power
+    (unaryfunc)unitAlgebraNegate,                // unaryfunc nb_negative
+    nullptr,                // unaryfunc nb_positive
+    nullptr,                // unaryfunc nb_absolute
+    nullptr,                // inquiry nb_nonzero (py2) nb_bool (py3)       /* Used by PyObject_IsTrue */
+    nullptr,                // unaryfunc nb_invert
+    nullptr,                // binaryfunc nb_lshift
+    nullptr,                // binaryfunc nb_rshift
+    nullptr,                // binaryfunc nb_and
+    nullptr,                // binaryfunc nb_xor
+    nullptr,                // binaryfunc nb_or
+    SST_NB_COERCE                // coercion nb_coerce Python 2 only
+    SST_NB_INTLONG((unaryfunc)unitAlgebraToLong) // unaryfunc nb_int (py1 nb_ling for py2)
+    SST_NB_RESERVED         // nb_reserved, py3 only
+    nullptr,                // unaryfunc nb_float
+    SST_NB_OCT                // unaryfunc nb_oct
+    SST_NB_HEX                // unaryfunc nb_hex
+    
+    /* Added in release 2.0 */
+    (binaryfunc)unitAlgebraAdd,  // binaryfunc nb_inplace_add
+    (binaryfunc)unitAlgebraSub,  // binaryfunc nb_inplace_subtract
+    (binaryfunc)unitAlgebraMul,  // binaryfunc nb_inplace_multiply
+    SST_NB_INPLACE_DIVIDE((binaryfunc)unitAlgebraDiv)   // binaryfunc nb_inplace_divide, py2 only
+    nullptr,                // binaryfunc nb_inplace_remainder
+    nullptr,                // ternaryfunc nb_inplace_power
+    nullptr,                // binaryfunc nb_inplace_lshift
+    nullptr,                // binaryfunc nb_inplace_rshift
+    nullptr,                // binaryfunc nb_inplace_and
+    nullptr,                // binaryfunc nb_inplace_xor
+    nullptr,                // binaryfunc nb_inplace_or
+    
+    /* Added in release 2.2 */
+    nullptr,                // binaryfunc nb_floor_divide
+    (binaryfunc)unitAlgebraDiv, // binaryfunc nb_true_divide
+    nullptr,                // binaryfunc nb_inplace_floor_divide
+    (binaryfunc)unitAlgebraDiv,  // binaryfunc nb_inplace_true_divide
+    
+    /* Added in release 2.5 */
+    nullptr,                // unaryfunc nb_index
+    SST_NB_MATRIX_MULTIPLY     // py3 only
+    SST_NB_INPLACE_MATRIX_MULTIPLY // py3 only
+}; 
+    
+// Other methods
+
+static PyObject* unitAlgebraGetRoundedValue(PyObject *self, PyObject *UNUSED(args))
+{
+    return unitAlgebraToLong(self);
```

Comment:
Does it make sense to also allow casting to float?

Reply:
Yes, I considered that but UnitAlgebra didn't support that cast.  However, looking at the code, the underlying data structure does have a toDouble, so we could make this work.